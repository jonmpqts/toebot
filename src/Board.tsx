import * as React from 'react';
import './Board.css';
import OpenSquare from './OpenSquare';
import OccupiedSquare from './OccupiedSquare';

interface Props {
    currentPlayer: 'X' | 'O';
    state: string;
    onSquareClick: (id: number) => void;
}

const Board = ({ currentPlayer, state, onSquareClick }: Props) => {
    const squares = state.split('').map((s, i) => {
        if (s === 'X') {
            return <OccupiedSquare key={i} occupiedBy="X" />;
        }

        if (s === 'O') {
            return <OccupiedSquare key={i} occupiedBy="O" />;
        }

        return (
            <OpenSquare
                key={i}
                currentPlayer={currentPlayer}
                onClick={() => onSquareClick(i)}
            />
        );
    });

    return (
        <div className="board">
            {squares}
        </div>
    );
};

export default Board;
