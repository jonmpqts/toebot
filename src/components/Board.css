.board {
    width: 450px;
}