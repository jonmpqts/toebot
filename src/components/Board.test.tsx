import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Board from './Board';

test('empty board', () => {
    const component = renderer.create(
        <Board
            currentPlayer="X"
            state="........."
            onSquareClick={(id) => { return; }}
        />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('second player move', () => {
    const component = renderer.create(
        <Board
            currentPlayer="O"
            state="X........"
            onSquareClick={(id) => { return; }}
        />
    );

    expect(component.toJSON()).toMatchSnapshot();
});
