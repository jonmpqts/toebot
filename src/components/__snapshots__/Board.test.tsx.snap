// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty board 1`] = `
<div
  className="board"
>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
</div>
`;

exports[`second player move 1`] = `
<div
  className="board"
>
  <div
    className="square X"
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </div>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-circle-o"
    />
  </button>
</div>
`;
